lambdas/check-mobile-app-vc-receipt/src/main/java/uk/gov/di/ipv/core/checkmobileappvcreceipt/CheckMobileAppVcReceiptHandler.java

```diff
@@ -183,8 +182,23 @@ private JourneyResponse getJourneyResponse(CheckMobileAppVcReceiptRequest reques
             return JOURNEY_ABANDON;
         }
 
+        var dcmawAsyncVc =
+                evcsService
+                        .getVerifiableCredentialsByState(
+                                userId, clientOAuthSessionItem.getEvcsAccessToken(), CURRENT)
+                        .get(CURRENT)
```

Comment:
After learning more about EVCS, I realise instead we'll want to fetch from PENDING_RETURN and load these VCs into the session credentials, if they exist.

```java
        ...

        if (CriResponseService.STATUS_ERROR.equals(criResponse.getStatus())) {
            return JOURNEY_ERROR;
        }
        if (CriResponseService.STATUS_ABANDON.equals(criResponse.getStatus())) {
            return JOURNEY_ABANDON;
        }

        var dcmawAsyncVc =
                evcsService
                        .getVerifiableCredentialsByState(
                                userId, clientOAuthSessionItem.getEvcsAccessToken(), CURRENT)
                        .get(CURRENT)
                        .stream()
                        .filter(vc -> DCMAW_ASYNC.equals(vc.getCri()))
                        .findFirst();

        if (dcmawAsyncVc.isEmpty()) {
            return null;
        }

        sessionCredentialsService.persistCredentials(
                List.of(dcmawAsyncVc.get()), auditEventUser.getSessionId(), false);

        ...
```